src/worker/helpers.go

```diff
@@ -166,46 +166,224 @@ func initOLDir(olPath string, dockerBaseImage string, newBase bool) (err error)
+	StoppedDirty
+	Unknown = -1
+)
+
+// Check the current state of Open Lambda
+//
+// This function returns the current state of Open Lambda, the PID if possible,
+// and an error if it encounters any.
+func checkState() (OlState, error) {
+	olPath := common.Conf.Worker_dir
+	dirStat, err := os.Stat(olPath)
+	if os.IsNotExist(err) {
+		// If OL Path doesn't exist, Open Lambda is not initialized.
+		return Uninitialized, nil
+	}
+	if !dirStat.IsDir() {
+		return Unknown, fmt.Errorf("%s is not a directory", olPath)
+	}
+
+	// Locate the worker.pid file, use it to get the worker's PID
+	pid, err := getPid()
+	if os.IsNotExist(err) {
+		// If we can't find the PID file, it probably means no OL instance is running.
+		return StoppedClean, nil
+	} else if err != nil {
+		// We will be in an unknown state if we encounter any other error.
+		return Unknown, fmt.Errorf("unexpected error occurred when reading PID file (%s)", err)
+	}
+
+	// On Unix systems, FindProcess always succeeds and returns a Process for the given PID,
+	// regardless of whether the process exists.
+	// https://pkg.go.dev/os#FindProcess
+	p, _ := os.FindProcess(pid)
+	if err := p.Signal(syscall.Signal(0)); err != nil {
+		// If we can't signal the process, it means the process isn't running and yet we found the PID file.
+		// Therefore, it was not cleanly shut down.
+		return StoppedDirty, nil
+	}
+
+	// If we can signal the process, it means the process is currently running.
+	return Running, nil
+}
+
+// Get the PID of the currently running OL instance.
+func getPid() (int, error) {
 	pidPath := filepath.Join(common.Conf.Worker_dir, "worker.pid")
-	data, err := ioutil.ReadFile(pidPath)
+	data, err := os.ReadFile(pidPath)
 	if os.IsNotExist(err) {
-		fmt.Printf("No worker appears to be running because %s does not exist.\n", pidPath)
-		return nil
+		return -1, os.ErrNotExist
 	} else if err != nil {
-		return err
+		return -1, fmt.Errorf("unexpected error occurred when reading PID file (%s)", err)
 	}
-	pidstr := string(data)
-	pid, err := strconv.Atoi(pidstr)
+	pidStr := string(data)
+	pid, err := strconv.Atoi(pidStr)
 	if err != nil {
-		return err
+		return -1, fmt.Errorf("unexpected error occurred when parsing PID file (%s) (%s)", pidStr, err)
 	}
+	return pid, nil
+}
 
-	fmt.Printf("According to %s, a worker should already be running (PID %d).\n", pidPath, pid)
-	p, err := os.FindProcess(pid)
+// This function will transition the Running state to StoppedClean state.
+// In other words, this function will stop the currently running OL instance.
+func runningToStoppedClean(olPath string) error {
+	fmt.Println("Attempting to gracefully shut down the worker process by sending SIGINT.")
+
+	pid, err := getPid()
 	if err != nil {
-		return fmt.Errorf("Failed to find worker process with PID %d.  May require manual cleanup.\n", pid)
+		fmt.Errorf("failed to get pid: %s", err)
 	}
-	fmt.Printf("Send SIGINT and wait for worker to exit cleanly.\n")
+
+	p, _ := os.FindProcess(pid)
```

Comment:
My understanding is that this should never return an error.  Let's write a little snippet to panic if we do get an error for some reason.